erpnext/utilities/bulk_transaction.py

```diff
@@ -0,0 +1,138 @@
+import json
+from datetime import date, datetime
+
+import frappe
+
+
+@frappe.whitelist()
+def transaction_processing(data, from_doctype, to_doctype):
+	if isinstance(data, str):
+		deserialized_data = json.loads(data)
+
+	else:
+		deserialized_data = data
+
+	length_of_data = len(deserialized_data)
+
+	if length_of_data > 10:
+		frappe.msgprint("Started a background job to create {1} {0}".format(to_doctype,length_of_data))
+		frappe.enqueue(job, deserialized_data=deserialized_data, from_doctype=from_doctype, to_doctype=to_doctype)
+	else:
+		job(deserialized_data, from_doctype, to_doctype)
+
+def job(deserialized_data, from_doctype, to_doctype):
+	failed_history = []
+	i = 0
+	for d in deserialized_data:
+		failed = []
+
+		try:
+			i+=1
+			doc_name = d.get('name')
+			task(doc_name, from_doctype, to_doctype)
+
+		except Exception as e:
+			failed_history.append(e)
+			failed.append(e)
+			update_logger(doc_name, e, from_doctype, to_doctype, status="Failed")
+
+		if not failed:
+			update_logger(doc_name, None, from_doctype, to_doctype, status="Success")
+
+	show_job_status(failed_history, deserialized_data, to_doctype)
+
+def task(doc_name, from_doctype, to_doctype):
+	from erpnext.accounts.doctype.payment_entry import payment_entry
+	from erpnext.accounts.doctype.purchase_invoice import purchase_invoice
+	from erpnext.accounts.doctype.sales_invoice import sales_invoice
+	from erpnext.buying.doctype.purchase_order import purchase_order
+	from erpnext.buying.doctype.supplier_quotation import supplier_quotation
+	from erpnext.selling.doctype.quotation import quotation
+	from erpnext.selling.doctype.sales_order import sales_order
+	from erpnext.stock.doctype.delivery_note import delivery_note
+	from erpnext.stock.doctype.purchase_receipt import purchase_receipt
+
+	mapper = {
+				"Sales Order": {"Sales Invoice": sales_order.make_sales_invoice, "Delivery Note": sales_order.make_delivery_note, "Advance Payment": payment_entry.get_payment_entry},
+				"Sales Invoice": {"Delivery Note": sales_invoice.make_delivery_note, "Payment": payment_entry.get_payment_entry},
+				"Delivery Note": {"Sales Invoice": delivery_note.make_sales_invoice, "Packing Slip": delivery_note.make_packing_slip},
+				"Quotation": {"Sales Order": quotation.make_sales_order, "Sales Invoice": quotation.make_sales_invoice},
+				"Supplier Quotation": {"Purchase Order": supplier_quotation.make_purchase_order, "Purchase Invoice": supplier_quotation.make_purchase_invoice, "Advance Payment": payment_entry.get_payment_entry},
+				"Purchase Order": {"Purchase Invoice": purchase_order.make_purchase_invoice, "Purchase Receipt": purchase_order.make_purchase_receipt},
+				"Purhcase Invoice": {"Purchase Receipt": purchase_invoice.make_purchase_receipt, "Payment": payment_entry.get_payment_entry},
+				"Purchase Receipt": {"Purchase Invoice": purchase_receipt.make_purchase_invoice}
+			}
+	if to_doctype == "Advance Payment" or to_doctype == "Payment":
```

Comment:
if to_doctype in ['Advance Payment', 'Payment']

Reply:
Did the changes